distance_metrics.py

```diff
@@ -0,0 +1,284 @@
+# Copyright (C) 2020 and later: Google, Inc.
+
+import cv2
+import enum
+import numpy as np
+
+class ImgFormat(enum.Enum):
+    """Enumeration class for image format."""
+    RGB = 1 # RGB format, necessary for ClearType
+    A8 = 2 # 8-bit grayscale format
+    A1 = 3 # black and white grayscale format
+    EMBEDDINGS = 4 # Arbitrary sized array for
+
+class Distance:
+    """Generator of distance metrics for specific image format.
+
+    Initialization:
+        >>> dis = Distance(img_format=ImgFormat.RGB)
+    Get available distance metrics for specific image format:
+        >>> metrics = dis.get_metrics()
+    'metrics' is a dictionary mapping from distance name to distance function：
+        >>> metrics.keys() # dict_keys(['manhattan', 'sum_squared', ...])
+    Calculate distance between two image arrays:
+        >>> metrics['manhattan'](img1, img2)
+    """
+
+    def __init__(self, img_format=ImgFormat.RGB):
+        """Store image format.
+
+        Args:
+            img_format: ImgFormat, format of input.
+        """
+
+        self.img_format = img_format
+
+    @property
+    def img_format(self):
+        """
+        Returns:
+                self._img_format: ImgFormat, format of input.
+        """
+        return self._img_format
+
+    @img_format.setter
+    def img_format(self, img_format):
+        """Raises TypeError if img_format is not a member of ImgFormat class."""
+        if img_format not in list(ImgFormat):
+            raise TypeError('Expect img_format to be a member of ImgFormat '
+                            'class.')
+        else:
+            self._img_format = img_format
+
+    def get_metrics(self):
+        """Return a dictionary of compatible distance names to functions.
+
+        Returns:
+            distances: Dict, mapping from names to distance functions.
+
+        Raises:
+            NotImplemented: if no functions implemented for current format.
+        """
+        if self.img_format == ImgFormat.RGB:
```

Comment:
since this function just return the metrics after each if block, it is better to simplified the structure from

if xxx:
  metrics = {  abc  }
elif yyy:
  metrics = {  efg  }
elif zzz:
  metrics = {  hij  }
else:
  klm
return metrics

to
if xxx:
  return { abc }
if yyy:
  return { efg }
if zzz:
  return { hij }
klm